sandbox/ide.py

```diff
@@ -0,0 +1,191 @@
+import os, shutil, subprocess, sys, yaml, socket
+from flask import Flask, request, url_for, redirect
+from werkzeug.security import gen_salt
+from nginx_utils import Server, Location
+
+from common.oauth_client import (
+    create_oauth_client,
+    is_staff,
+    login,
+    get_user,
+)
+from common.rpc.hosted import add_domain
+from common.shell_utils import sh
+
+HOSTNAME = "cs61a.org"
+SANDBOX = f"sb.{HOSTNAME}" if HOSTNAME == "cs61a.org" else HOSTNAME
+CODESERVER = f"ide.{HOSTNAME}" if HOSTNAME == "cs61a.org" else HOSTNAME
+
+app = Flask(__name__)
+
+create_oauth_client(app, "61a-sandbox")
+
+
+@app.route("/")
+def index():
+    if not is_staff("cs61a"):
+        return login()
+
+    username = get_user()["email"].split("@")[0]
```

Comment:
Enforce that it's an `@berkeley.edu` to avoid conflicts.

```
        if not is_staff_userdata(ret.json()["data"]) or not g.email.endswith(
            "@berkeley.edu"
        ):
            raise PermissionError
        g.username = (
            g.email[: -len("@berkeley.edu")] if is_prod_build() else DEFAULT_USER
        )
```
from the old sandbox tool.